drivers/Aqara/aqara-lock/src/test/test_aqara_lock.lua

```diff
@@ -0,0 +1,153 @@
+      { credentialId = 2, credentialType = "fingerprint", userId = "2", userLabel = "user2", userType = "regularUser" }
+    }
+    mock_device:set_field(HOST_COUNT, 0, { persist = true })
+    test.socket.capability:__queue_receive({ mock_device.id,
+      { capability = lockCredentialInfo.ID, component = "main", command = "syncAll", args = credentialInfoData }
+    })
+    test.socket.capability:__expect_send(mock_device:generate_test_message("main",
+      remoteControlStatus.remoteControlEnabled('true', { visibility = { displayed = false } })))
+    test.socket.capability:__expect_send(mock_device:generate_test_message("main",
+      lockCredentialInfo.credentialInfo(credentialInfoData_copy, { visibility = { displayed = false } })))
+  end
+)
+
+test.register_coroutine_test(
+  "credential_utils.upsert_credential_info(host user)",
+  function()
+    local credentialInfoData = {
+      { credentialId = 1, credentialType = "keypad", userId = "1", userLabel = "user1", userType = "host" }
+    }
+    test.socket.capability:__queue_receive({ mock_device.id,
+      { capability = lockCredentialInfo.ID, component = "main", command = "upsert", args = { credentialInfoData } } })
+    test.socket.capability:__expect_send(mock_device:generate_test_message("main",
+      remoteControlStatus.remoteControlEnabled('true', { visibility = { displayed = false } })))
+    test.socket.capability:__expect_send(mock_device:generate_test_message("main",
+      lockCredentialInfo.credentialInfo(credentialInfoData, { visibility = { displayed = false } })))
+  end
+)
+
+test.register_coroutine_test(
+  "credential_utils.upsert_credential_info(regular user)",
+  function()
+    local credentialInfoData = {
+      { credentialId = 1, credentialType = "keypad", userId = "1", userLabel = "user1", userType = "regularUser" }
+    }
+    test.socket.capability:__queue_receive({ mock_device.id,
+      { capability = lockCredentialInfo.ID, component = "main", command = "upsert", args = credentialInfoData } })
+  end
+)
+
+test.register_coroutine_test(
+  "credential_utils.delete_user",
+  function()
+    test.socket.capability:__queue_receive({ mock_device.id,
+      { capability = lockCredentialInfo.ID, component = "main", command = "deleteUser", args = { userId = "3" } } })
+  end
+)
+
+test.register_coroutine_test(
+  "credential_utils.delete_credential",
+  function()
+    test.socket.capability:__queue_receive({ mock_device.id,
+      { capability = lockCredentialInfo.ID, component = "main", command = "deleteCredential", args = { credentialId = "1" } } })
+  end
+)
+
+test.register_coroutine_test(
+  "lock_state_handler - shared_key is nil",
+  function()
+    local attr_report_data = {
+      { PRI_ATTR, data_types.OctetString.ID, "\x93\x15\xAA\xFE\x78\xEE\x3E\x81\x9A\x06\xE3\x9A\x62\xD3\xB1\xF1\xD4\x64\x8C\x16\x66\xAB\xE1\x69\x89\x8C\x04\x56\x8D\xAD\xEA\xDE\xF8" }
```

Comment:
This seems a bit magical, if this test started failing it would be hard to know why, is there any way to provide the context of what this message is supposed to represent?

Reply:
This is the sample messages sent by the door lock to generate a shared key.
This message consists of a command("\x93") and the sample public key("\x15 .. ") of a doorlock.